plugins/destination/plugin.go

```diff
@@ -83,12 +83,8 @@ type Plugin struct {
 	// Logger to call, this logger is passed to the serve.Serve Client, if not define Serve will create one instead.
 	logger zerolog.Logger
 
-	// This is in use if the user passed a managed client
-	metrics     map[string]*Metrics
-	metricsLock *sync.RWMutex
-
-	workers     map[string]*worker
-	workersLock *sync.Mutex
+	sync.RWMutex // implies usage by pointer
```

Comment:
Could we rather not use an embedding here? It's arguably nicer to read `p.Lock()`, but it makes it harder for a reader to figure out where the `[R]Lock`, `[R]Unlock` functions are defined, and it doesn't tell them what the lock is supposed to be for. I believe it also now makes `Lock` / `Unlock` public functions on `Plugin`; probably not something we want as a public API.

Reply:
I'll move it to `workersMux` then